rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictFluentReturnRector/Fixture/multi_return_this.php.inc

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromStrictFluentReturnRector\Fixture;
+
+final class MultiReturnThis
+{
+    private $foo = 'bar';
+
+    public function test()
+    {
+        if (rand(0, 1)) {
+            $this->foo = 'foo';
+            return $this;
+        }
+
+        return $this;
+    }
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromStrictFluentReturnRector\Fixture;
+
+final class MultiReturnThis
+{
+    private $foo = 'bar';
+
+    public function test(): self
+    {
+        if (rand(0, 1)) {
+            $this->foo = 'foo';
+            return $this;
```

Comment:
Another test with a `return $this` and `return null` case would be great to support nullable

Reply:
that's already covered on https://getrector.com/demo/cda3b293-ac7d-4432-83f5-f23cc833a4c1 while return `?static` on final class is not needed, that should be `?self`, but that will need separate PR for improvement and consideration.

Reply:
I created PR:

- https://github.com/rectorphp/rector-src/pull/4891

for nullable `$this` return on `ReturnUnionTypeRector`